### DISCUSSION TITLE
[Craft 4] Relative dates for date conditions

### DISCUSSION THREAD

**Discussion by thupsi** [craftcms/cms#10749] [Ideas] [answered]
I've been testing the condition builder in the Craft 4 beta and I like it a lot, but, as would be expected for a new feature, it has some room to grow. 

One thing I'm missing for example is the ability to set a date condition that is relative to the current date.

So, in addition to this:
**FROM static date - TO static date**
we could have something like this:
 ****FROM x days** before/after current date - **TO x days** before/after current date.**

We could also have some presets, like 'last week', 'today' etc, but that would be just an added convienience.

I'm thinking this would be especially useful in custom sources.

**edit:** 
Even better would be something like:
****FROM: `more than/exactly/less than` `x` days** `before/after` current date - **TO: `more than/exactly/less than` `x`  days  `before/after`  current date.****

**Comment by shifuma** [comment]
Yes! This would be an amazing addition, exactly what I need.

**Comment by brandonkelly** [comment, marked as answer]
This has been addressed for Craft 4.3 via #11888. Going forward, date range condition rules have the following options:

- Today
- This week
- This month
- This year
- Past 7 days
- Past 30 days
- Past 90 days
- Past year
- Before… [relative date]
- After… [relative date]
- Range… [from and to dates]
- has a value
- is empty

**Reply by brandonkelly** [reply]
Craft 4.3 was released earlier this week, with these additions!
